### PR TITLE
Vite: Fix stats-plugin to normalize file names with posix paths

### LINKED ISSUE #27172: [Bug]: Chromatic TurboSnap incompatible with preview-stats.json generated from builder-vite
### Describe the bug

In Storybook 8 the [vite-plugin-turbosnap](https://github.com/IanVS/vite-plugin-turbosnap) package was ported into `@storybook/builder-vite`

I had previously reported an [issue](https://github.com/IanVS/vite-plugin-turbosnap/issues/14) with `vite-plugin-turbosnap` as on Windows it generates files paths for `name` and `id` using the system path separators (`\\`) instead of posix separators (`/`).

With Webpack, even on Windows, `name` and `id` always use posix separators. This has been assumed to always be the case for the chromatic-cli tool.

With the current preview-stats.json file generated by builder-vite, it is incompatible with turbosnap in Chromatic.

To fix this, the `normalize` function in [webpack-stats-plugin.ts](https://github.com/storybookjs/storybook/blob/next/code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts) needs to be modified to convert `workingDir` and `filename` to posix paths, using the same posix transformer as chromatic-cli.

```diff
export function pluginWebpackStats({ workingDir }: WebpackStatsPluginOptions): WebpackStatsPlugin {
+ // Replaces Windows-style backslash path separators with POSIX-style forward slashes, because the
+ // Webpack stats use forward slashes in the `name` and `moduleName` fields. Note `changedFiles`
+ // already contains forward slashes, because that's what git yields even on Windows.
+ const posix = (localPath: string) => localPath.split(path.sep).filter(Boolean).join(path.posix.sep);
  /**
   * Convert an absolute path name to a path relative to the vite root, with a starting `./`
   */
  function normalize(filename: string) {
    // Do not try to resolve virtual files
    if (filename.startsWith('/virtual:')) {
      return filename;
    }
    // Otherwise, we need them in the format `./path/to/file.js`.
    else {
+     const posixWorkingDir = posix(workingDir);
+     const posixFilename = posix(filename);

+     const relativePath = path.posix.relative(posixWorkingDir, stripQueryParams(posixFilename));
-     const relativePath = path.relative(workingDir, stripQueryParams(filename));
      // This seems hacky, got to be a better way to add a `./` to the start of a path.
      return `./${relativePath}`;
    }
  }
```

### To Reproduce

1. Create a storybook project using builder-vite
2. Run `storybook build --stats-json` on a Window environment
3. Check the `name` and `id` fields in `storybook-static/preview-stats.json` for Windows path separators

### System

```bash
Storybook Environment Info:

  System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
  Binaries:
    Node: 18.18.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - C:\Program Files\nodejs\yarn.CMD
    npm: 9.8.1 - C:\Program Files\nodejs\npm.CMD <----- active
  Browsers:
    Edge: Chromium (124.0.2478.105)
  npmPackages:
    @storybook/addon-a11y: ^8.0.10 => 8.0.10
    @storybook/addon-essentials: ^8.0.10 => 8.0.10
    @storybook/addon-interactions: ^8.0.10 => 8.0.10
    @storybook/addon-links: ^8.0.10 => 8.0.10
    @storybook/addon-onboarding: ^8.0.10 => 8.0.10
    @storybook/addon-viewport: ^8.0.10 => 8.0.10
    @storybook/blocks: ^8.0.10 => 8.0.10
    @storybook/builder-vite: ^8.0.10 => 8.0.10
    @storybook/manager-api: ^8.0.10 => 8.0.10
    @storybook/react: ^8.0.10 => 8.0.10
    @storybook/react-vite: ^8.0.10 => 8.0.10
    @storybook/test: ^8.0.10 => 8.0.10
    @storybook/test-runner: ^0.18.0 => 0.18.0
    @storybook/theming: ^8.0.10 => 8.0.10
    chromatic: ^10.5.0 => 10.9.6
    eslint-plugin-storybook: ^0.8.0 => 0.8.0
    msw-storybook-addon: ^2.0.2 => 2.0.2
    storybook: ^8.0.10 => 8.0.10
    storybook-addon-remix-react-router: ^3.0.0 => 3.0.0
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AlexAtVista**
Closes #27172

## What I did

Updates the `normalize` function within the `webpack-stats-plugin` for `builder-vite` to always normalize file names with posix path separators (`/`). 

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. On a window environment, create a storybook sandbox using react vite
2. Build storybook using `storybook build --stats-json`
3. Confirm the `name` and `id` fields in the `preview-stats.json` file are using posix path separators (`/`) and not windows separators (`\\`)

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts [hunk 1]
```diff
@@ -56,7 +59,9 @@ export function pluginWebpackStats({ workingDir }: WebpackStatsPluginOptions): W
     }
     // Otherwise, we need them in the format `./path/to/file.js`.
     else {
-      const relativePath = path.relative(workingDir, stripQueryParams(filename));
+      const posixWorkingDir = posix(workingDir);
```
Comment (shilman) [inline code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts:62]: We're already prebundling `slash` into this package, so can we use that instead?

https://www.npmjs.com/package/slash

**Review by shilman — CHANGES_REQUESTED** [review]

File: code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts (hunk 1 above)
Comment (AlexAtVista) [reply code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts]: @shilman great suggestion, updated

**Review by shilman — APPROVED** [review]
🙌

File: code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts [hunk 2]
```diff
@@ -58,7 +59,7 @@ export function pluginWebpackStats({ workingDir }: WebpackStatsPluginOptions): W
     else {
       const relativePath = path.relative(workingDir, stripQueryParams(filename));
```
Comment (valentinpalkovic) [inline code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts:60]: Couldn't we just use `path.posix.relative` here to use forward slashes, independently of the OS?

Comment (AlexAtVista) [reply code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts:60]: @valentinpalkovic that is how I originally submitted this PR, until @shilman suggested using `slash` to be consistent with how windows paths are already converted elsewhere. I don't mind either way, if you're able to decide between you which option is preferable?

Comment (valentinpalkovic) [reply code/builders/builder-vite/src/plugins/webpack-stats-plugin.ts:60]: I see. Sorry that I haven’t followed the whole conversation! It’s fine then how it is!

**Pull request merged**